### PR TITLE
overwrite to() for QTensor and QBitsTensor

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ? 
This PR fixes the `to()` method so that it is applied to the tensors `_data`, `_scale` and `_zeropoint` when using `QTensor` or `QBitsTensor`. Before this PR, doing the following `QLinear().to(device)` would not change the device of these tensors.

I ran the following tests (that were not passing before) :` python -m pytest -sv test/nn/test_qlinear.py::test_move_qlinear`

I need to check why these tests are not passing anymore with this PR :
```
FAILED test/tensor/ops/test_quantized_dispatch.py::test_to_device[cuda] - AssertionError: assert 'cpu' == 'cuda'
FAILED test/tensor/ops/test_quantized_dispatch.py::test_softmax[cuda-5-5-1] - AssertionError
FAILED test/tensor/ops/test_quantized_dispatch.py::test_softmax[cuda-5-5-10] - AssertionError
FAILED test/tensor/ops/test_quantized_dispatch.py::test_softmax[cuda-32-32-1] - AssertionError
FAILED test/tensor/ops/test_quantized_dispatch.py::test_softmax[cuda-32-32-10] - AssertionError
FAILED test/tensor/ops/test_quantized_dispatch.py::test_softmax[cuda-10-32-1] - AssertionError
FAILED test/tensor/ops/test_quantized_dispatch.py::test_softmax[cuda-10-32-10] - AssertionError
```

cc @dacorvo

### CODE REVIEW

File: test/tensor/test_qtensor.py [hunk 1]
```diff
@@ -8,6 +8,13 @@
 from quanto import QTensor, absmax_scale
 
 
+def test_qtensor_move(device):
+    input_shape = (2, 4, 8)
+    qa = random_qtensor(input_shape, dtype=torch.float32)
+    qa = qa.to(device)
+    assert qa._data.device.type == device.type
+    assert qa._scale.device.type == device.type
```
Comment (SunMarc) [inline test/tensor/test_qtensor.py:16]: This works even before this PR. This is why you never had this issue.

File: quanto/tensor/core.py [hunk 2]
```diff
@@ -313,6 +313,10 @@ def __torch_dispatch__(cls, op, types, args, kwargs=None):
     def numpy(self):
         return self.dequantize().cpu().numpy()
 
+    def to(self, *args, **kwargs):
+        self._data = self._data.to(*args, **kwargs)
+        self._scale = self._scale.to(*args, **kwargs)
+        return self
```
Comment (SunMarc) [inline quanto/tensor/core.py:319]: I wanted to return `super().to(*args, **kwargs)` but it was causing weird behavior with tests using `QBitsTensor` and it was calling `__torch_function__` after . To reproduce, return `super().to(*args, **kwargs)`  and run `python -m pytest -sv test/nn/test_qlinear.py::test_move_qlinear`

**Comment by dacorvo** [comment]
I am not sure there is actually an issue with QTensor, as .to() is already correctly dispatched here:

https://github.com/huggingface/quanto/blob/6302171b7569a3fd86f31e1731d01d390d1eb557/quanto/tensor/ops.py#L74

It won't work for QBitsTensor that has an extra zeropoint inner tensor, but it should be fixed at the dispatch stage, and not directly in the class declaration IMHO.

File: test/tensor/test_qtensor.py (hunk 1 above)
Comment (dacorvo) [reply test/tensor/test_qtensor.py:16]: There is already a test for this here: https://github.com/huggingface/quanto/blob/6302171b7569a3fd86f31e1731d01d390d1eb557/test/tensor/ops/test_quantized_dispatch.py#L8

File: quanto/tensor/core.py (hunk 2 above)
Comment (dacorvo) [reply quanto/tensor/core.py:306]: Tensor subclasses are a very special beast: you should not override the base Tensor methods that way, and instead do it through the dispatch.

**Pull request closed without merging**